File: Server/src/main/content/global/skill/production/cooking/recipe/topping/impl/TunaAndCorn.java
package content.global.skill.production.cooking.recipe.topping.impl;

import content.global.skill.production.cooking.recipe.Recipe;
import core.game.interaction.NodeUsageEvent;
import core.game.node.entity.player.Player;
import core.game.node.entity.skill.Skills;
import core.game.node.item.Item;

/**
 * The Tuna and corn.
 */
public final class TunaAndCorn extends Recipe {

    private static final Item CHOPPED_TUNA = new Item(7086);

    private static final Item COOKED_CORN = new Item(5988);

    private static final Item TUNA_AND_CORN = new Item(7068);

    @Override
    public void mix(final Player player, final NodeUsageEvent event) {
        if (player.getSkills().getLevel(Skills.COOKING) < 67) {
            player.getDialogueInterpreter().sendDialogue("You need a Cooking level of at least " + 57 + " in order to do this.");
            return;
        }
        super.mix(player, event);
        player.getSkills().addExperience(Skills.COOKING, 204, true);
    }

    @Override
    public Item getBase() {
        return CHOPPED_TUNA;
    }

    @Override
    public Item getProduct() {
        return TUNA_AND_CORN;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{COOKED_CORN};
    }

    @Override
    public Item[] getParts() {
        return new Item[]{};
    }

    @Override
    public String getMixMessage(NodeUsageEvent event) {
        return null;
    }

    @Override
    public boolean isSingular() {
        return true;
    }

}

File: Server/src/main/content/global/skill/production/cooking/recipe/topping/impl/TunaAndCorn.kt
package content.global.skill.production.cooking.recipe.topping.impl

import content.global.skill.production.cooking.recipe.Recipe
import core.api.consts.Items
import core.api.getStatLevel
import core.api.rewardXP
import core.api.sendDialogue
import core.game.interaction.NodeUsageEvent
import core.game.node.entity.player.Player
import core.game.node.entity.skill.Skills
import core.game.node.item.Item

class TunaAndCorn : Recipe() {

    override fun mix(player: Player, event: NodeUsageEvent) {
        if (getStatLevel(player, Skills.COOKING) < 67) {
            sendDialogue(player, "You need a Cooking level of at least 57 in order to do this.")
            return
        }
        super.mix(player, event)
        rewardXP(player, Skills.COOKING, 204.0)
    }

    override fun getBase(): Item {
        return CHOPPED_TUNA
    }

    override fun getProduct(): Item {
        return TUNA_AND_CORN
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(COOKED_CORN)
    }

    override fun getParts(): Array<Item> {
        return emptyArray()
    }

    override fun getMixMessage(event: NodeUsageEvent): String? {
        return null
    }

    override fun isSingular(): Boolean {
        return true
    }

    companion object {
        private val CHOPPED_TUNA = Item(Items.CHOPPED_TUNA_7086)
        private val COOKED_CORN = Item(Items.COOKED_SWEETCORN_5988)
        private val TUNA_AND_CORN = Item(Items.TUNA_AND_CORN_7068)
    }
}

